hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -455,23 +453,26 @@ public void onStateInitialized(
                 readableStore.getFreezeTime(),
                 readableStore.getLastFrozenTime());
 
-        HederaSoftwareVersion deserializedVersion = null;
+        ServicesSoftwareVersion deserializedVersion = null;
         // We do not support downgrading from one version to an older version.
-        if (previousVersion instanceof HederaSoftwareVersion hederaSoftwareVersion) {
-            deserializedVersion = hederaSoftwareVersion;
-            if (version.isBefore(deserializedVersion)) {
-                logger.fatal(
-                        "Fatal error, state source version {} is higher than node software version {}",
-                        deserializedVersion,
-                        version);
-                throw new IllegalStateException("Cannot downgrade from " + deserializedVersion + " to " + version);
-            }
+        if (previousVersion instanceof ServicesSoftwareVersion servicesSoftwareVersion) {
+            deserializedVersion = servicesSoftwareVersion;
+        } else if (previousVersion instanceof HederaSoftwareVersion hederaSoftwareVersion) {
+            deserializedVersion = new ServicesSoftwareVersion(
+                    hederaSoftwareVersion.servicesVersion(), hederaSoftwareVersion.configVersion());
         } else {
             if (previousVersion != null) {
                 logger.fatal("Deserialized state not created with Hedera software");
                 throw new IllegalStateException("Deserialized state not created with Hedera software");
             }
         }
+        if (version.compareTo(deserializedVersion) < 0) {
```

Comment:
Makes sense to keep this outside this case `previousVersion instanceof HederaSoftwareVersion hederaSoftwareVersion` 👍